scripts/marketing_data_test.py

```diff
@@ -0,0 +1,451 @@
+    actions_before = negative_df['action_id'].value_counts()
+    actions_after = df_downsampled['action_id'].value_counts()
+    logger.info(f'Before: {actions_before}')
+    logger.info(f'After: {actions_after}')
+
+    # Final sample
+    combo_df = pd.concat([positive_df, df_downsampled], axis=0).reset_index(drop=True)
+    # this shuffles it
+    combo_df = combo_df.sample(frac=1).reset_index(drop=True)
+    X = combo_df[X.columns.values]
+    T_feat = combo_df[T_feat.columns.values]
+    y = combo_df['y'].to_numpy()
+    t_id = combo_df['action_id'].to_numpy()
+
+    n_pos_ex2 = y.sum()
+    logger.info(f"After downsampling non-conversion, Conversion rate  now {n_pos_ex2/len(y)}")
+    logger.info(f"X {X.shape}")
+    logger.info(f"T {T_feat.shape}")
+    logger.info(f"y {y.shape}")
+
+    return X, y, T_feat, t_id
+
+
+def marketing_dml_test(
+    file_name: str,
+    model_name: str,
+    cv_folds: int = 2,
+    collinearity_thresh: float = 0.5,
+    downsample_ratio: float = 0.1,
+    log_scale_skew_thresh: float | int = 3,
+    dml_cv_params: str | None = None,
+    y_cv_params: str | None = None,
+    t_cv_params: str | None = None,
+    val_pcnt: float = 0.2,
+    do_evaluate: bool = False
+):
+    """
+    Main function for testing DML on high dimensional marketing data sets. Adds pre-processing
+    like scaling and removing highly collinear features. It can cross validates parameters of either
+    first stage model, or the final model.  Saves out the scores of every model to a file.
+
+    :param file_name: File that contains the data, assumed to be joblib
+    :param model_name: Name of the model to use, either SparseLinearDML or CausalForestDML
+    :param cv_folds: Number of CV folds to use in the DML algorithm
+    :param collinearity_thresh: Collinearity threshold for removing collinear features
+    :param downsample_ratio: Ratio to downsample non-positive examles
+    :param log_scale_skew_thresh: Threshold on skew to employ log scaling
+    :param dml_cv_params: Dict-as-string CV params for the final stage model
+    :param y_cv_params: Dict-as-string CV params for the first stage outcome model
+    :param t_cv_params: Dict-as-string CV params for the first stage treatment model
+
+    """
+    dml_cv_param_dict = json.loads(dml_cv_params) if dml_cv_params is not None else {}
+    y_cv_param_dict = json.loads(y_cv_params) if y_cv_params is not None else {}
+    t_cv_param_dict = json.loads(t_cv_params) if t_cv_params is not None else {}
+
+    root_dir = os.environ.get("LOCAL_ANALYSIS_DIR", ".")
+    logger.info(f"Using root dir {root_dir} loading data file {file_name}")
+    data_file = os.path.join(os.path.abspath(root_dir), file_name)
+    data = joblib.load(data_file)
+
+    # This is specific to Offerfit - names of the data are derived from contextual bandits
+    t_id = data['action_ids']
+    X = data['X_no_act']
+    T_feat = data['a_processed']
+    y = data['real_reward']
+
+    treat_df = pd.DataFrame(T_feat)
```

Comment:
Based on my local debugger, T_feat is already a pd.DataFrame. Do we need to make it a dataframe again?